components/brave_shields/browser/ad_block_service.cc

```diff
@@ -66,27 +66,36 @@ void AdBlockService::ShouldStartRequest(
     const GURL& url,
     blink::mojom::ResourceType resource_type,
     const std::string& tab_host,
+    bool aggressive_blocking,
     bool* did_match_rule,
     bool* did_match_exception,
     bool* did_match_important,
     std::string* mock_data_url) {
-  AdBlockBaseService::ShouldStartRequest(
-      url, resource_type, tab_host, did_match_rule, did_match_exception,
-      did_match_important, mock_data_url);
-  if (did_match_important && *did_match_important) {
-    return;
+  if (aggressive_blocking ||
+      base::FeatureList::IsEnabled(
+          brave_shields::features::kBraveAdblockDefault1pBlocking) ||
+      !SameDomainOrHost(
+          url,
+          url::Origin::CreateFromNormalizedTuple("https", tab_host.c_str(), 80),
```

Comment:
no need for `c_str()`

Reply:
what are our expectations if `tab_host` is empty? AFAIR this is the SW case?

Reply:
also, just curious whether we want to skip the check for all of the subresource types? Or maybe just scripts/whatever else?

Reply:
removed `c_str()`

this is the same logic that's passed to the adblock engine already. It should be possible to refactor it to only happen in a single place later, but at the moment it's not trivial.

I'll leave @pes10k to comment on the resource types we want to apply this to.

Reply:
I see this change is outdated, so apologies if this is talking into the void, but i dont see any reason to not apply to all requests.  The "what does shields block in what configurations" logic is already complicated as it is.  I dont think we should make the decision tree more complicated unless there is a compelling reason